src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesArray = sourceString
+    .split(';').filter(style =>
+      style.trim() !== '');
+
+  const stylesObject = stylesArray.reduce(function(result, style) {
+    const parts = style.split(':').map(part => part.trim());
```

Comment:
you can destruct property from part: `const [property, value] = style.split(':').map(part => part.trim());`

and then you can remove redundant:

```
const property = parts[0];
const value = parts[1];
```